controllers/awsmachine_controller.go

```diff
@@ -419,17 +437,28 @@ func (r *AWSMachineReconciler) findInstance(scope *scope.MachineScope, ec2svc se
 	return instance, nil
 }
 
-func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *scope.MachineScope, clusterScope cloud.ClusterScoper, ec2Scope scope.EC2Scope, elbScope scope.ELBScope) (ctrl.Result, error) {
+//nolint:gocyclo // Temporary.
+func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *scope.MachineScope, clusterScope cloud.ClusterScoper, ec2Scope scope.EC2Scope, elbScope scope.ELBScope, objectStoreScope scope.S3Scope) (ctrl.Result, error) {
 	machineScope.Info("Reconciling AWSMachine")
 
+	var objectStoreSvc services.ObjectStoreInterface
+
+	if objectStoreScope != nil {
+		objectStoreSvc = r.getObjectStoreService(objectStoreScope)
+	}
+
 	// If the AWSMachine is in an error state, return early.
 	if machineScope.HasFailed() {
 		machineScope.Info("Error state detected, skipping reconciliation")
 
-		if machineScope.UseSecretsManager() {
```

Comment:
IIRC, this is because `UseSecretsManager()` now requires to know what user data format is being used, which is only available when the bootstrap data became available and that might not be the case when the machine gets removed right after it was created.

Calling `deleteEncryptedBootstrapDataSecret()` should also be safe at any point as far as I tested, as the secrets backend is always available and no secrets will be actually removed as the secrets counter will remaing to be 0 until needed.

Reply:
Got it, probably need to block users to specify secretCount when insecureSkipSecretsManager is true with webhooks. This was not an issue when we used UseSecretsManager().

Reply:
There is already this:
```go
func (r *AWSMachine) validateCloudInitSecret() field.ErrorList {
  var allErrs field.ErrorList

  if r.Spec.CloudInit.InsecureSkipSecretsManager {
    if r.Spec.CloudInit.SecretPrefix != "" {
      allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "cloudInit", "secretPrefix"), "cannot be set if spec.cloudInit.insecureSkipSecretsManager is true"))
    }
    if r.Spec.CloudInit.SecretCount != 0 {
      allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "cloudInit", "secretCount"), "cannot be set if spec.cloudInit.insecureSkipSecretsManager is true"))
    }
```

Do we need this logic also for AWSMachineTemplate?